packages/arb-validator-core/ethbridge/arbRollupWatcher.go

```diff
@@ -40,46 +42,63 @@ import (
-var transactionMessageDeliveredID ethcommon.Hash
-var ethDepositMessageDeliveredID ethcommon.Hash
-var depositERC20MessageDeliveredID ethcommon.Hash
-var depositERC721MessageDeliveredID ethcommon.Hash
-var contractTransactionMessageDeliveredID ethcommon.Hash
+var transactionID ethcommon.Hash
+var transactionBatchID ethcommon.Hash
+var ethDepositID ethcommon.Hash
+var depositERC20ID ethcommon.Hash
+var depositERC721ID ethcommon.Hash
+var contractTxID ethcommon.Hash
+var transactionBatchTxCallABI abi.Method
+
+type TransactionBatchTxCallArgs struct {
+	Chain       ethcommon.Address
+	Tos         []ethcommon.Address
+	SeqNumbers  []*big.Int
+	Values      []*big.Int
+	DataLengths []uint32
+	Data        []byte
+	Signatures  []byte
+}
 
 func init() {
 	parsedRollup, err := abi.JSON(strings.NewReader(rollup.ArbRollupABI))
 	if err != nil {
 		panic(err)
 	}
-	parsedInbox, err := abi.JSON(strings.NewReader(rollup.IGlobalInboxABI))
+	inbox, err := abi.JSON(strings.NewReader(globalinbox.GlobalInboxABI))
 	if err != nil {
 		panic(err)
 	}
 	rollupCreatedID = parsedRollup.Events["RollupCreated"].ID()
-	rollupStakeCreatedID = parsedRollup.Events["RollupStakeCreated"].ID()
-	rollupChallengeStartedID = parsedRollup.Events["RollupChallengeStarted"].ID()
-	rollupChallengeCompletedID = parsedRollup.Events["RollupChallengeCompleted"].ID()
+	stakeCreatedID = parsedRollup.Events["RollupStakeCreated"].ID()
+	challengeStartedID = parsedRollup.Events["RollupChallengeStarted"].ID()
+	challengeCompletedID = parsedRollup.Events["RollupChallengeCompleted"].ID()
 	rollupRefundedID = parsedRollup.Events["RollupStakeRefunded"].ID()
 	rollupPrunedID = parsedRollup.Events["RollupPruned"].ID()
 	rollupStakeMovedID = parsedRollup.Events["RollupStakeMoved"].ID()
 	rollupAssertedID = parsedRollup.Events["RollupAsserted"].ID()
 	rollupConfirmedID = parsedRollup.Events["RollupConfirmed"].ID()
 	confirmedAssertionID = parsedRollup.Events["ConfirmedAssertion"].ID()
 
-	transactionMessageDeliveredID = parsedInbox.Events["TransactionMessageDelivered"].ID()
-	ethDepositMessageDeliveredID = parsedInbox.Events["EthDepositMessageDelivered"].ID()
-	depositERC20MessageDeliveredID = parsedInbox.Events["ERC20DepositMessageDelivered"].ID()
-	depositERC721MessageDeliveredID = parsedInbox.Events["ERC721DepositMessageDelivered"].ID()
-	contractTransactionMessageDeliveredID = parsedInbox.Events["ContractTransactionMessageDelivered"].ID()
+	transactionID = inbox.Events["TransactionMessageDelivered"].ID()
+	transactionBatchID = inbox.Events["TransactionMessageBatchDelivered"].ID()
+	ethDepositID = inbox.Events["EthDepositMessageDelivered"].ID()
+	depositERC20ID = inbox.Events["ERC20DepositMessageDelivered"].ID()
+	depositERC721ID = inbox.Events["ERC721DepositMessageDelivered"].ID()
+	contractTxID = inbox.Events["ContractTransactionMessageDelivered"].ID()
+
+	transactionBatchTxCallABI = inbox.Methods["deliverTransactionBatch"]
+	transactionBatchTxCallABI.Inputs[5].Name = "data"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [avoid magic numbers like '5', create a named constant for it](https://app.codacy.com/gh/OffchainLabs/arbitrum/pullRequest?prid=5428340)